spec/support/match_xml.rb

```diff
@@ -0,0 +1,36 @@
+def match_xml(expected, ignore_list)
+  MatchXml.new(expected, ignore_list)
+end
+
+class MatchXml
+  def initialize(expected, ignore_list)
+    @ignore_list = ignore_list
+    @expected = MatchXml.xml_to_css_path_hash(expected, @ignore_list)
+  end
+
+  def matches?(actual)
+    @actual = MatchXml.xml_to_css_path_hash(actual, @ignore_list)
+    RSpec::Matchers::BuiltIn::Include.new(@expected).matches?(@actual)
+  end
+
+  def failure_message
+    RSpec::Support::Differ.new.diff(@actual, @expected)
+  end
+
+  class << self
+    def traverse_elements(base_node, ignore_list, &block)
+      return to_enum(:traverse_elements, base_node, ignore_list) unless block_given?
+      return if ignore_list.include?(base_node.name)
+      base_node.element_children.each do |child_node|
```

Comment:
comparing here instead of before the iteration over children means we don't compare the root, and might mean that we can only check `ignore_list` once. probably move it out